panic in mondoo shell if i use k8s.pod.podSpec

I get panic in my ` mondoo shell k8s` if I use ` k8s.pod.podSpec`. I expect something like: resource not usable without discovery flag. 



````mondoo shell k8s
→ loaded configuration from /home/user/.config/mondoo/mondoo.yml using source default
→ discover related assets for 1 asset(s)
→ use cluster name from kube config cluster-name=arn:aws:eks:us-east-2:921877552404:cluster/patrick-container-escape-demo-udb3-cluster
→ resolved assets resolved-assets=1
                        .-.            
                        : :            
,-.,-.,-. .--. ,-.,-. .-' : .--.  .--. ™
: ,. ,. :' .; :: ,. :' .; :' .; :' .; :
:_;:_;:_;`.__.':_;:_;`.__.'`.__.'`.__.'  interactive shell

mondoo> k8s.pod.podSpec
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5ba73e2]

goroutine 257 [running]:
go.mondoo.com/cnquery/resources/packs/k8s.initNamespacedResource[...](0xc0004e4210, 0x7827120, 0x82a3f90?)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/resources/packs/k8s/common.go:232 +0x462
go.mondoo.com/cnquery/resources/packs/k8s.(*mqlK8sPod).init(0xc000f5b5d0?, 0x40f4df?)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/resources/packs/k8s/pod.go:59 +0x30
go.mondoo.com/cnquery/resources/packs/k8s.newK8sPod(0xc001ec73b0, 0x0?)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/resources/packs/k8s/k8s.lr.go:2344 +0xcf
go.mondoo.com/cnquery/resources.(*Runtime).CreateResourceWithID(0xc001ec73b0, {0xc00087f137, 0x7}, {0x0, 0x0}, {0xcbacc70, 0x0, 0x0})
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/resources/runtime.go:155 +0x1c5
go.mondoo.com/cnquery/resources.(*Runtime).CreateResource(...)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/resources/runtime.go:186
go.mondoo.com/cnquery/llx.(*blockExecutor).createResource(0xc0003204d0, {0xc00087f137, 0x7}, 0x7?, 0x1?)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/llx/llx.go:652 +0xbf
go.mondoo.com/cnquery/llx.(*blockExecutor).runGlobalFunction(0xc0003204d0, 0xc000c4e5a0, 0xc9ede60, 0x7f66a00b29f0?)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/llx/llx.go:697 +0x2e5
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc00186c9d8?, 0xc000f5ba38?, 0x8a00000000000008?)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/llx/llx.go:723 +0x130
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc0003204d0, 0xc000f5ba58?, 0x52e6d4c?)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/llx/llx.go:754 +0x2a8
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc00185d290?, 0x6ee3760?)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/llx/llx.go:779 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc0003204d0, 0x7e9535a?)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/llx/llx.go:811 +0xd4
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc0003204d0)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/llx/llx.go:337 +0x2d7
go.mondoo.com/cnquery/llx.(*MQLExecutorV2).Run(0xc00187c4e0?)
        /home/user/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20220930134759-6bdcc58a0fef/llx/llx.go:269 +0x59
go.mondoo.com/mondoo/policy/executor/internal.(*executionManager).executeCodeBundle(0xc000c4e6e0, 0xc001ec5400, 0x408bd1?, {0x0, 0x0})
        /home/user/go/src/go.mondoo.io/mondoo/policy/executor/internal/execution_manager.go:168 +0x365
go.mondoo.com/mondoo/policy/executor/internal.(*executionManager).Start.func1()
        /home/user/go/src/go.mondoo.io/mondoo/policy/executor/internal/execution_manager.go:87 +0x1cb
created by go.mondoo.com/mondoo/policy/executor/internal.(*executionManager).Start
        /home/user/go/src/go.mondoo.io/mondoo/policy/executor/internal/execution_manager.go:56 +0x6a
```

--- a/resources/packs/k8s/common.go
+++ b/resources/packs/k8s/common.go
@@ -227,6 +227,10 @@ func initNamespacedResource[T K8sNamespacedObject](
 		}
 	}
 
+	if matchFn == nil {
+		return args, *new(T), fmt.Errorf("cannot use resource without specifying id or name/namespace")
+	}
+
 	for i := range nsResources {
 		nsR := nsResources[i].(T)
 		if matchFn(nsR) {
@@ -288,6 +292,10 @@ func initResource[T K8sObject](
 			name, _ := nsR.Name()
 			return name == nameRaw
 		}
+	}
+
+	if matchFn == nil {
+		return args, *new(T), fmt.Errorf("cannot use resource without specifying id or name")
 	}
 
 	for i := range resources {

